storage/src/lsm_tree/components/sstable.rs

```diff
@@ -152,21 +152,22 @@ impl SstableMeta {
         buf.put_u64_le(self.data_size as u64);
         let checksum = crc32sum(&buf[4..]);
         (&mut buf[..4]).put_u32_le(checksum);
-        buf.freeze()
+        buf
     }
 
-    pub fn decode(mut buf: Bytes) -> Self {
-        let checksum = buf.get_u32_le();
+    pub fn decode(buf: Vec<u8>) -> Self {
+        let mut rbuf = &buf[..];
```

Comment:
Is it okay to simply use `let mut buf = &buf[..]` here? I'm not sure if it's okay.

Reply:
`BlockMeta::decode` takes in a reference of mut Buf, I think it's ok?

Reply:
how about change the signature to `pub fn decode(buf: &mut &[u8]) `?